pkg/homedir/homedir_unix.go

```diff
@@ -101,9 +101,39 @@ func isWriteableOnlyByOwner(perm os.FileMode) bool {
 	return (perm & 0o722) == 0o700
 }
 
-// GetConfigHome returns XDG_CONFIG_HOME.
+// ConfigHome returns XDG_CONFIG_HOME.
+// ConfigHome returns $HOME/.config and nil error if XDG_CONFIG_HOME is not set.
+// Verifies ownership of config directory matches the current process or
+// returns error.
+// See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
+func GetConfigHome() (string, error) {
+	rootlessConfigHomeDirOnce.Do(func() {
+		tmpDir := os.Getenv("XDG_CONFIG_HOME")
+		if tmpDir == "" {
+			home := Get()
+			resolvedHome, err := filepath.EvalSymlinks(home)
+			if err != nil {
+				rootlessConfigHomeDirError = fmt.Errorf("cannot resolve %s: %w", home, err)
+				return
+			}
+			tmpDir = filepath.Join(resolvedHome, ".config")
+		}
+		st, err := os.Stat(tmpDir)
+		if err == nil {
+			if int(st.Sys().(*syscall.Stat_t).Uid) == os.Geteuid() {
+				rootlessConfigHomeDir = tmpDir
+			} else {
+				rootlessConfigHomeDirError = fmt.Errorf("path %q exists and it is not owned by the current user", tmpDir)
+				return
+			}
+		}
+	})
+
+	return rootlessConfigHomeDir, rootlessConfigHomeDirError
+}
+
+// GetConfigHome returns XDG_CONFIG_HOME. (Deprecated)
```

Comment:
If we go with the plan of gradual migration per https://github.com/containers/podman/issues/23818#issuecomment-2329502508 , please
- point at the alternative
- explain why it is deprecated / what needs to change in callers
- consider using the standard `Deprecated: ` syntax that linters understand. OTOH that would force us to migrate immediately, I’m not 100% sure we want to sign up for doing that